wallet/cardano-wallet.js

```diff
@@ -188,37 +185,38 @@ const CardanoWallet = (secretOrMnemonic, CARDANOLITE_CONFIG) => {
   }
 
   async function getChangeAddress(usedAddressesLimit = Number.MAX_SAFE_INTEGER, offset = 0) {
-    const usedAddressesAndSecrets = await getUsedAddressesAndSecrets()
+    const usedAddressesWithHdNodes = await getUsedAddressesWithHdNodes()
 
     let result
 
-    if (usedAddressesAndSecrets.length < usedAddressesLimit) {
+    if (usedAddressesWithHdNodes.length < usedAddressesLimit) {
       const highestUsedChildIndex = Math.max(
         0x8000000,
-        ...usedAddressesAndSecrets.map((item) => item.childIndex)
+        ...usedAddressesWithHdNodes.map((item) => item.childIndex)
       )
 
-      result = (await getAddressAndSecret(highestUsedChildIndex + 1 + offset)).address
+      result = (await getAddressWithHdNode(highestUsedChildIndex + 1 + offset)).address
     } else {
       result =
-        usedAddressesAndSecrets[Math.floor(Math.random() * usedAddressesAndSecrets.length)].address
+        usedAddressesWithHdNodes[Math.floor(Math.random() * usedAddressesWithHdNodes.length)]
```

Comment:
Just a question -- pseudorandom generator is safe here?

Reply:
It's just a simple heuristics, if You run off the new address pool, pick one used randomly, true rnd won't help it much.